src/server/pps/server/monitor.go

```diff
@@ -420,15 +437,9 @@ func (m *ppsMaster) makeCronCommits(ctx context.Context, in *pps.Input) error {
 		return err // Shouldn't happen, as the input is validated in CreatePipeline
 	}
 	pachClient := m.a.env.GetPachClient(ctx)
-	// make sure there isn't an unfinished commit on the branch
-	commitInfo, err := pachClient.InspectCommit(in.Cron.Repo, "master", "")
-	if err != nil {
+	// finish any open commit on the branch
+	if err := pachClient.FinishCommit(in.Cron.Repo, "master", ""); err != nil && !pfs_server.IsCommitFinishedErr(err) {
 		return err
-	} else if commitInfo != nil && commitInfo.Finished == nil {
-		// and if there is, delete it
-		if err = pachClient.SquashCommitSet(commitInfo.Commit.ID); err != nil {
-			return err
-		}
```

Comment:
some part of this block is wrong - the if condition isn't sufficient to avoid losing data, but it didn't seem clear to me that we should squash the entire commitset, potentially deleting unrelated data, just because a user (accidentally?) opened a commit on the cron repo. This could use broader discussion, as maybe `squash` should have more safeguards even after we fix the data loss bug.